<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1020</width>
    <height>720</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1020</width>
    <height>720</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1020</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>OPA Soundchip editor - beta</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="op1Box">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>421</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>Operator 1</string>
    </property>
    <widget class="OperatorWidget" name="op1Widget" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>16</y>
       <width>420</width>
       <height>260</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="op2Box">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>10</y>
      <width>421</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>Operator 2</string>
    </property>
    <widget class="OperatorWidget" name="op2Widget" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>16</y>
       <width>420</width>
       <height>260</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="op3Box">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>310</y>
      <width>421</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>Operator 3</string>
    </property>
    <widget class="OperatorWidget" name="op3Widget" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>16</y>
       <width>420</width>
       <height>260</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="op4Box">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>310</y>
      <width>421</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>Operator 4</string>
    </property>
    <widget class="OperatorWidget" name="op4Widget" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>420</width>
       <height>260</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="programBox">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>10</y>
      <width>141</width>
      <height>421</height>
     </rect>
    </property>
    <property name="title">
     <string>Program</string>
    </property>
    <widget class="QPushButton" name="i1Push">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>1</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="i2Push">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>2</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="i3Push">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>3</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="i4Push">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>4</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="i6Push">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>6</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="i7Push">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>7</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="i8Push">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>8</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="i5Push">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>5</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="ProgramWidget" name="programWidget" native="true">
     <property name="geometry">
      <rect>
       <x>-1</x>
       <y>140</y>
       <width>141</width>
       <height>281</height>
      </rect>
     </property>
    </widget>
    <widget class="Led" name="i1Led" native="true">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>20</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
    <widget class="Led" name="i2Led" native="true">
     <property name="geometry">
      <rect>
       <x>48</x>
       <y>20</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
    <widget class="Led" name="i3Led" native="true">
     <property name="geometry">
      <rect>
       <x>78</x>
       <y>20</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
    <widget class="Led" name="i4Led" native="true">
     <property name="geometry">
      <rect>
       <x>108</x>
       <y>20</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
    <widget class="Led" name="i5Led" native="true">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>80</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
    <widget class="Led" name="i6Led" native="true">
     <property name="geometry">
      <rect>
       <x>48</x>
       <y>80</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
    <widget class="Led" name="i7Led" native="true">
     <property name="geometry">
      <rect>
       <x>78</x>
       <y>80</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
    <widget class="Led" name="i8Led" native="true">
     <property name="geometry">
      <rect>
       <x>108</x>
       <y>80</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>440</y>
      <width>141</width>
      <height>151</height>
     </rect>
    </property>
    <property name="title">
     <string>Global</string>
    </property>
    <widget class="QLabel" name="masterTuneLabel">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>20</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Tune</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QDial" name="masterTuneDial">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>40</y>
       <width>49</width>
       <height>49</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-128</number>
     </property>
     <property name="maximum">
      <number>127</number>
     </property>
     <property name="notchTarget">
      <double>16.000000000000000</double>
     </property>
     <property name="notchesVisible">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="masterTuneEditLabel">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>90</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::TextEditorInteraction</set>
     </property>
    </widget>
    <widget class="QLabel" name="masterVolumeLabel">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Volume</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QDial" name="masterVolumeDial">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>49</width>
       <height>49</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="notchTarget">
      <double>16.000000000000000</double>
     </property>
     <property name="notchesVisible">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="masterVolumeEditLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::TextEditorInteraction</set>
     </property>
    </widget>
    <widget class="QPushButton" name="stealingButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>115</y>
       <width>121</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Voice stealing</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="algo1Push">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-a.png</normaloff>:/images/res/program-a.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo2Push">
    <property name="geometry">
     <rect>
      <x>85</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-b.png</normaloff>:/images/res/program-b.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo3Push">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-c.png</normaloff>:/images/res/program-c.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo4Push">
    <property name="geometry">
     <rect>
      <x>235</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-d.png</normaloff>:/images/res/program-d.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo5Push">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-e.png</normaloff>:/images/res/program-e.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo6Push">
    <property name="geometry">
     <rect>
      <x>385</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-f.png</normaloff>:/images/res/program-f.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo7Push">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-g.png</normaloff>:/images/res/program-g.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo8Push">
    <property name="geometry">
     <rect>
      <x>535</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-h.png</normaloff>:/images/res/program-h.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo11Push">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-k.png</normaloff>:/images/res/program-k.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo12Push">
    <property name="geometry">
     <rect>
      <x>835</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-l.png</normaloff>:/images/res/program-l.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo10Push">
    <property name="geometry">
     <rect>
      <x>685</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-j.png</normaloff>:/images/res/program-j.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo9Push">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-i.png</normaloff>:/images/res/program-i.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="algo13Push">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>600</y>
      <width>71</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/images/res/program-m.png</normaloff>:/images/res/program-m.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>64</width>
      <height>64</height>
     </size>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1020</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="fileMenu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="quitAction"/>
   </widget>
   <widget class="QMenu" name="connectionMenu">
    <property name="title">
     <string>Connection</string>
    </property>
   </widget>
   <widget class="QMenu" name="helpMenu">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="aboutAction"/>
   </widget>
   <widget class="QMenu" name="deviceMenu">
    <property name="title">
     <string>Device</string>
    </property>
    <addaction name="loadProgramAction"/>
    <addaction name="sendProgramAction"/>
    <addaction name="separator"/>
    <addaction name="allNotesOffAction"/>
    <addaction name="allSoundsOffAction"/>
    <addaction name="separator"/>
    <addaction name="memoryProtectionAction"/>
   </widget>
   <addaction name="fileMenu"/>
   <addaction name="connectionMenu"/>
   <addaction name="deviceMenu"/>
   <addaction name="helpMenu"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="quitAction">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="aboutAction">
   <property name="text">
    <string>About ...</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="loadProgramAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Load program</string>
   </property>
  </action>
  <action name="sendProgramAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Send program</string>
   </property>
  </action>
  <action name="allNotesOffAction">
   <property name="text">
    <string>All notes off</string>
   </property>
  </action>
  <action name="allSoundsOffAction">
   <property name="text">
    <string>All sounds off</string>
   </property>
  </action>
  <action name="memoryProtectionAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Memory protection</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OperatorWidget</class>
   <extends>QWidget</extends>
   <header>operatorwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ProgramWidget</class>
   <extends>QWidget</extends>
   <header>programwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Led</class>
   <extends>QWidget</extends>
   <header>led.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
